ui/page/main_page.go

```diff
@@ -553,9 +553,10 @@ func (mp *MainPage) Layout(gtx layout.Context) layout.Dimensions {
 }
 
 func (mp *MainPage) LayoutUSDBalance(gtx layout.Context) layout.Dimensions {
+	mp.UpdateBalance()
```

Comment:
Looking carefully, i dont think this should be in the `LayoutUSDBalance` reason is this is meant for just the USD balance display.

What happens if it is moved to the `Layout`?